src/gui/optionsdlg.cpp

```diff
@@ -1182,6 +1206,9 @@ bool OptionsDialog::useAdditionDialog() const
 void OptionsDialog::enableApplyButton()
 {
     applyButton->setEnabled(true);
+
+    // Verify if the share action button must be enabled
+    m_ui->comboRatioLimitAct->setEnabled(m_ui->checkMaxRatio->isChecked() || m_ui->checkMaxSeedingMinutes->isChecked());
```

Comment:
I did try it and understand what it does otherwise i wouldn't make a comment. I meant that you put it in `enableApplyButton()` which is executed when pretty much anything is changed in the options.